pipeline.py

```diff
@@ -72,31 +101,120 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+
+        for article in articles:
+            raw_text = article.get_raw_text()
+            processed_tokens = self._process(raw_text)
+
+        cleaned_tokens = []
+        single_tagged_tokens = []
+        multiple_tagged_tokens = []
+
+        for processed_token in processed_tokens:
+            cleaned_tokens.append(processed_token.get_cleaned())
+            single_tagged_tokens.append(processed_token.get_single_tagged())
+            multiple_tagged_tokens.append(processed_token.get_multiple_tagged())
+
+        article.save_as(' '.join(cleaned_tokens), kind=ArtifactType.cleaned)
+        article.save_as(' '.join(single_tagged_tokens), kind=ArtifactType.single_tagged)
+        article.save_as(' '.join(multiple_tagged_tokens), kind=ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # txt from pdf comes with words like след-ующий
+        # this replace deals with them
+        text = raw_text.replace('-\n', '')
+
+        result = Mystem().analyze(text)
+
+        morph_tokens = []
+
+        for token in result:
+
+            # pre requisites for the token to be usable
+            if "analysis" not in token:
+                continue
+            if not token["analysis"]:
+                continue
+
+            original_word = token["text"]
+            if not re.match(r"[A-Za-zА-Яа-яЁё]", original_word):
```

Comment:
1. pattern in loop
2. why do we need it? if it is for an English language is there any other more elegant way of understanding that? For example, by examining mystem report structure for each analyzed word?

Reply:
1. understood
2. There could be English names withing the Russian text, I wanted to account for them without additional structures. But as I recall now, we shouldn't just analyze them, I will delete it.